File: src/com/juhnowski/onthology/TotalityOfAllFacts.java
package com.juhnowski.onthology;

public class TotalityOfAllFacts {
    public Facts facts = new Facts();

    private static volatile TotalityOfAllFacts instance;

    public static TotalityOfAllFacts getInstance() {
        TotalityOfAllFacts localInstance = instance;
        if (localInstance == null) {
            synchronized (TotalityOfAllFacts.class) {
                localInstance = instance;
                if (localInstance == null) {
                    instance = localInstance = new TotalityOfAllFacts();
                }
            }
        }
        return localInstance;
    }
}

File: src/com/juhnowski/onthology/TotalityOfAllFacts.kt
package com.juhnowski.onthology

class TotalityOfAllFacts {
    var facts = Facts()

    companion object {

        @Volatile
        private var instance: TotalityOfAllFacts? = null

        fun getInstance(): TotalityOfAllFacts {
            var localInstance = instance
            if (localInstance == null) {
                synchronized(TotalityOfAllFacts::class.java) {
                    localInstance = instance
                    if (localInstance == null) {
                        localInstance = TotalityOfAllFacts()
                        instance = localInstance
                    }
                }
            }
            return localInstance
        }
    }
}
